src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/RestoreArgs.cs

```diff
@@ -174,7 +174,7 @@ public void ApplyStandardProperties(RestoreRequest request)
                 request.LockFilePath = ProjectJsonPathUtilities.GetLockFilePath(request.Project.FilePath);
             }
 
-            if (request.Project.RestoreMetadata != null) {
+            if (request.Project.RestoreMetadata?.CacheFilePath == null) {
```

Comment:
@DoRonMotter 
RestoreMetadata will never be null in productive code at this point. It's actually already been used prior to this call. 

I'm cleaning this up now.